internal/post-comments/post-comments.go

```diff
@@ -0,0 +1,68 @@
+package postcomments
+
+import (
+	"grab/internal/comment"
+	"grab/internal/post"
+)
+
+type PostWithCommentsResponse struct {
+	Posts []PostWithComments `json:"posts"`
+}
+
+type PostWithComments struct {
+	ID       int64     `json:"id"`
+	Title    string    `json:"string"`
+	Comments []comment.Comment `json:"comments,omitempty"`
+}
+
+type PostCommentsWorker interface {
+	GetPostWithComments() (PostWithCommentsResponse, error)
+	combinePostWithComments([]post.Post, []comment.Comment) []PostWithComments
+}
+
+type PostCommentsWorkerImp struct {}
+
+// CommentWorker Constructor
+func CreatePostCommentsWorker() *PostCommentsWorkerImp{
+	return &PostCommentsWorkerImp{}
+}
+func (pwc *PostCommentsWorkerImp) GetPostWithComments() (PostWithCommentsResponse, error) {
+
+	postwker := post.CreatePostWorker()
+	posts, error := postwker.GetPosts()
+	if error != nil {
+		return PostWithCommentsResponse{}, error
```

Comment:
`return nil, error` is good enough, no need to return `PostWithCommentsResponse{}` every time you got error